### PR TITLE
fix: Use the name of the leftmost expression in horizontal operations

### PR DESCRIPTION
**Pull request by reswqa**
As we discussed before, expression shouldn't have a specific name.

### CODE REVIEW

**Comment by stinodego** [comment]
Ritchie recently changed this for `any/all`. It makes sense to do it for the others too.

It is technically a breaking change but I suppose we could see it as a bug fix.

**Comment by ritchie46** [comment]
Yes. If we didn't it was a bug. Thank you @reswqa

File: crates/polars-ops/src/series/ops/horizontal.rs [hunk 1]
```diff
@@ -1,34 +1,14 @@
...
-                    .try_reduce_with(|l, r| sum_fn(&l, &r).map(Cow::Owned))
-                    // we can unwrap the option, because we are certain there is a series
-                    .unwrap()
-                    .map(|cow| cow.into_owned())
-            })
-        },
-    };
-    out.map(|ok| ok.with_name("sum"))
+pub fn sum_horizontal(s: &[Series]) -> PolarsResult<Option<Series>> {
+    let df = DataFrame::new_no_checks(Vec::from(s));
```
Comment (ritchie46) [inline crates/polars-ops/src/series/ops/horizontal.rs:9]: I am not sure we can do this. We do support broadcasting behavior. Is that covered here?

Comment (ritchie46) [reply crates/polars-ops/src/series/ops/horizontal.rs:9]: Yes.

Comment (ritchie46) [reply crates/polars-ops/src/series/ops/horizontal.rs:9]: Cool thanks!

**Pull request merged**